spec/models/queries/projects/project_query_spec.rb

```diff
@@ -387,4 +387,66 @@
       end
     end
   end
+
+  describe "scopes" do
+    describe ".public_lists" do
+      it "returns only public lists" do
+        public_query = create(:project_query, public: true)
+        public_query_other_user = create(:project_query, public: true)
+        create(:project_query, public: false)
+
+        expect(described_class.public_lists).to contain_exactly(public_query, public_query_other_user)
+      end
+    end
+
+    describe ".private_lists" do
+      it "returns only private lists owned by the user" do
+        create(:project_query, public: true)
+        private_query = create(:project_query, public: false)
+        create(:project_query, public: false)
+
+        expect(described_class.private_lists(user: private_query.user)).to contain_exactly(private_query)
+      end
+    end
+
+    describe ".visible" do
+      it "returns public and private queries owned by the user" do
+        public_query = create(:project_query, public: true)
+        public_query_other_user = create(:project_query, public: true)
+        private_query = create(:project_query, public: false)
+        create(:project_query, public: false)
```

Comment:
Nitpick - it may be easier to understand if all instances for all scopes tests are created together at the top, also second public query is not really needed in tests, as it is not different from the first one

Reply:
Yes, it is different as it belongs to another user. So we are testing that the owner is irrelevant when queries are marked as public

Reply:
Both don't have user explicitly set, and neither `public_query.user`, nor `public_query_other_user.user` are used, so they are just two public queries without relation to other queries